### PR TITLE
gh-94619: Remove long deprecated methods module_repr() and load_module()

### LINKED ISSUE #94619: Remove previously deprecated importlib functions
**Bug report**

* `module_repr()` was [deprecated](https://docs.python.org/3.11/library/importlib.html?highlight=namespaceloader#importlib.abc.Loader.module_repr) since Python 3.4 and is slated for removal in [Python 3.12](https://github.com/python/cpython/blob/main/Lib/importlib/_bootstrap_external.py#L1354) as per the warnings, so let's remove them.  See also #86302
* Similarly, `load_module()` has also been [deprecated](https://docs.python.org/3.11/library/importlib.html?highlight=namespaceloader#importlib.abc.Loader.load_module) since Python 3.4, so let's try removing it too.

### PR DESCRIPTION
**Pull request by warsaw**
Closes #94619

* Issue: gh-94619

### CODE REVIEW

**Comment by warsaw** [comment]
@brettcannon ping

File: Misc/NEWS.d/next/Library/2022-07-06-14-57-33.gh-issue-94619.PRqKVX.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove long deprecated methods `module_repr()` and `load_module()`.
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Library/2022-07-06-14-57-33.gh-issue-94619.PRqKVX.rst]:
```suggestion
Remove the long-deprecated `module_repr()` from `importlib`.
```

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by warsaw** [comment]
I have made the requested changes; please review again.

**Comment by warsaw** [comment]
@brettcannon re-ping!

**Review by brettcannon — APPROVED** [review]

**Pull request merged**